cmd/clusterctl/client/repository/repository_versions.go

```diff
@@ -39,11 +39,16 @@ func latestContractRelease(repo Repository, contract string) (string, error) {
 	}
 	// Attempt to check if the latest release satisfies the API Contract
 	// This is a best-effort attempt to find the latest release for an older API contract if it's not the latest release.
-	// If an error occurs, we just return the latest release.
 	file, err := repo.GetFile(latest, metadataFile)
+	// If an error occurs, we just return the latest release.
 	if err != nil {
+		if errors.Is(err, errNotFound) {
```

Comment:
Wouldn't it be better to move this wrapping to inside GetFile when we get at error after calling `getReleaseByTag` ([here](https://github.com/ykakarap/cluster-api/blob/05bf57fd606c1bb0a9b570ccb4b1269f603af0eb/cmd/clusterctl/client/repository/repository_github.go#L159-L161))?

Reply:
Seems more appropriate there to show that the release was not found over doing it when error wrapping for a "not found" case of `GetFile`

Reply:
I'm sorry for  the slow feedback loop.

I moved the wrapping into GetFile, but had to keep the errors.Is here, because otherwise the error would be ignored 👍